RealmSwift/Tests/ProjectionTests.swift

```diff
@@ -0,0 +1,1792 @@
+    @Projected(\ModernAllTypesObject.mapOptInt64) var mapOptInt64
+    @Projected(\ModernAllTypesObject.mapOptFloat) var mapOptFloat
+    @Projected(\ModernAllTypesObject.mapOptDouble) var mapOptDouble
+    @Projected(\ModernAllTypesObject.mapOptString) var mapOptString
+    @Projected(\ModernAllTypesObject.mapOptBinary) var mapOptBinary
+    @Projected(\ModernAllTypesObject.mapOptDate) var mapOptDate
+    @Projected(\ModernAllTypesObject.mapOptDecimal) var mapOptDecimal
+    @Projected(\ModernAllTypesObject.mapOptObjectId) var mapOptObjectId
+    @Projected(\ModernAllTypesObject.mapOptUuid) var mapOptUuid
+
+    @Projected(\ModernAllTypesObject.linkingObjects) var linkingObjects
+}
+
+class AdvancedObject: Object {
+    @Persisted(primaryKey: true) var pk: ObjectId
+    @Persisted var commonArray: RealmSwift.List<Int>
+    @Persisted var objectsArray: RealmSwift.List<SimpleObject>
+    @Persisted var commonSet: MutableSet<Int>
+    @Persisted var objectsSet: MutableSet<SimpleObject>
+}
+
+extension SimpleObject {
+    var stringify: String {
+        "\(int) - \(bool)"
+    }
+}
+
+class AdvancedProjection: Projection<AdvancedObject> {
+    @Projected(\AdvancedObject.commonArray.count) var arrayLen
+    @Projected(\AdvancedObject.commonArray) var renamedArray
+    @Projected(\AdvancedObject.objectsArray.projectTo.stringify) var projectedArray: ProjectedCollection<String>
+    @Projected(\AdvancedObject.commonSet.first) var firstElement
+    @Projected(\AdvancedObject.objectsSet.projectTo.bool) var projectedSet: ProjectedCollection<Bool>
+}
+
+class FailedProjection: Projection<ModernAllTypesObject> {
+    @Projected(\ModernAllTypesObject.ignored) var ignored
+}
+
+public class AddressSwift: EmbeddedObject {
+    @Persisted var city: String = ""
+    @Persisted var country = ""
+}
+
+public class CommonPerson: Object {
+    @Persisted var firstName: String
+    @Persisted var lastName = ""
+    @Persisted var birthday: Date
+    @Persisted var address: AddressSwift?
+    @Persisted public var friends: RealmSwift.List<CommonPerson>
+    @Persisted var reviews: RealmSwift.List<String>
+    @Persisted var money: Decimal128
+}
+
+@available(OSX 10.15, watchOS 6.0, iOS 13.0, iOSApplicationExtension 13.0, OSXApplicationExtension 10.15, tvOS 13.0, *)
+public final class PersonProjection: Projection<CommonPerson> {
+    @Projected(\CommonPerson.firstName) var firstName
+    @Projected(\CommonPerson.lastName.localizedUppercase) var lastNameCaps
+    @Projected(\CommonPerson.birthday.timeIntervalSince1970) var birthdayAsEpochtime
+    @Projected(\CommonPerson.address?.city) var homeCity
+    @Projected(\CommonPerson.friends.projectTo.firstName) var firstFriendsName: ProjectedCollection<String>
+}
+
+public class SimpleObject: Object {
+    @Persisted var int: Int
+    @Persisted var bool: Bool
+}
+
+@available(OSX 10.15, watchOS 6.0, iOS 13.0, iOSApplicationExtension 13.0, OSXApplicationExtension 10.15, tvOS 13.0, *)
```

Comment:
What about these require a higher deployment target?

Reply:
not required. fixed.